modin/engines/base/io/text/text_file_dispatcher.py

```diff
@@ -258,3 +259,157 @@ def _read_rows(
+        header: int, list of int or str ("infer")
+                Original header parameter of read_csv function.
+        names:  array
+                Original names parameter of read_csv function.
+        Returns
+        -------
+        header_size:
+                The number of rows that are used by header.
+        """
+        header_size = 0
+        if header == "infer" and names is None:
+            header_size += 1
+        elif isinstance(header, int):
+            header_size += header + 1
+        elif hasattr(header, "__iter__") and not isinstance(header, str):
+            header_size += max(header) + 1
+
+        return header_size
+
+    @classmethod
+    def _define_usecols(
+        cls,
+        filepath_or_buffer: FilePathOrBuffer,
+        **kwargs,
+    ):
+        """
+        Validate usecols parameter and convert it to integer type
+        if it is not integer yet.
+        ----------
+        filepath_or_buffer:
+                parameter, which defines data location.
+        kwargs:
+                read_csv kwargs.
+
+        Returns
+        -------
+        usecols:
+                validated/converted usecols parameter.
+        """
+        usecols = kwargs.get("usecols", None)
+        usecols_md = _validate_usecols_arg(usecols)
+        if usecols is not None and usecols_md[1] != "integer":
+            all_cols = pandas.read_csv(
+                cls.file_open(filepath_or_buffer, "rb"),
+                **dict(kwargs, nrows=0, skipfooter=0),
+            ).columns
+            usecols = all_cols.get_indexer_for(list(usecols_md[0]))
+
+        return usecols
+
+    @classmethod
+    def _define_metadata(
+        cls,
+        df: pandas.DataFrame,
+        num_splits: int,
+        column_names: Sequence,
+    ):
+        """
+        Define partitioning metadata.
+        ----------
+        df:
+            The DataFrame to split.
+        num_splits:
+            The maximum number of splits to separate the DataFrame into.
+        column_names:
+            column names of df.
+
+        Returns
+        -------
+        column_widths:
+                Column width to use during new frame creation (number of
+                columns for each partition).
+        num_splits:
+                Updated `num_splits` parameter.
+        """
+        column_chunksize = compute_chunksize(df, num_splits, axis=1)
+
+        if column_chunksize > len(column_names):
+            column_widths = [len(column_names)]
+            # This prevents us from unnecessarily serializing a bunch of empty
+            # objects.
+            num_splits = 1
+        else:
+            column_widths = [
+                column_chunksize
+                if len(column_names) > (column_chunksize * (i + 1))
+                else 0
+                if len(column_names) < (column_chunksize * i)
+                else len(column_names) - (column_chunksize * i)
+                for i in range(num_splits)
+            ]
```

Comment:
Would this be more readable if we used normal if statements rather than inline if statements?

Reply:
Yes, i think, it would, regular `if else` statements were used.